test/util/indexing.jl

```diff
@@ -0,0 +1,181 @@
+    groups_eight = UnitRange{UInt8}[1:5, 6:10, 11:15, 16:20, 21:25]
+    int_index_one = Indexer(symbolnames)
+    int_index_two = Indexer(textnames, UInt64)
+    int_index_three = Indexer(symbolnames, UInt8)
+    g_index_one = Indexer(symbolnames, groups)
+    g_index_two = Indexer(textnames, groups)
+    g_index_three = Indexer(textnames, groups_eight)
+    empty_index = Indexer(UInt64)
+
+    @testset "Correct Construction" begin
+        @test int_index_one == int_index_two
+        @test int_index_one == int_index_three
+        @test int_index_two == int_index_three
+        @test g_index_one == g_index_two
+        @test g_index_one == g_index_two
+        @test g_index_two == g_index_three
+        @test int_index_one != empty_index
+        @test g_index_one != empty_index
+        @test typeof(int_index_one) != typeof(int_index_two)
+        @test typeof(int_index_one) != typeof(int_index_three)
+        @test typeof(int_index_two) != typeof(int_index_three)
+        @test typeof(g_index_one) == typeof(g_index_two)
+        @test typeof(g_index_one) != typeof(g_index_three)
+        @test typeof(g_index_two) != typeof(g_index_three)
+        @test typeof(int_index_one) == Indexer{Int64}
+        @test typeof(int_index_two) == Indexer{UInt64}
+        @test typeof(int_index_three) == Indexer{UInt8}
+        @test typeof(g_index_one) == Indexer{UnitRange{Int64}}
+        @test typeof(g_index_two) == Indexer{UnitRange{Int64}}
+        @test typeof(g_index_three) == Indexer{UnitRange{UInt8}}
+        @test typeof(empty_index) == Indexer{UInt64}
+    end
+
+    @testset "Basic Operators" begin
+        @test length(int_index_one) == 5
+        @test length(int_index_two) == 5
+        @test length(int_index_three) == 5
+        @test length(g_index_one) == 5
+        @test length(g_index_two) == 5
+        @test length(g_index_three) == 5
+        @test names(int_index_one) == symbolnames
+        @test names(int_index_two) == symbolnames
+        @test names(int_index_three) == symbolnames
+        @test names(g_index_one) == symbolnames
+        @test names(g_index_two) == symbolnames
+        @test names(g_index_three) == symbolnames
+        @test names(empty_index) == Symbol[]
+        @test isequal(copy(int_index_one), int_index_one) && !is(copy(int_index_one), int_index_one)
+        @test isequal(copy(int_index_two), int_index_two) && !is(copy(int_index_two), int_index_two)
+        @test isequal(copy(int_index_three), int_index_three) && !is(copy(int_index_three), int_index_three)
+        @test isequal(copy(g_index_one), g_index_one) && !is(copy(g_index_one), g_index_one)
+        @test isequal(copy(g_index_two), g_index_two) && !is(copy(g_index_two), g_index_two)
+        @test isequal(copy(g_index_three), g_index_three) && !is(copy(g_index_three), g_index_three)
+        @test isequal(copy(empty_index), empty_index) && !is(copy(empty_index), empty_index)
+        for n in symbolnames
```

Comment:
Pedantry: This seems like it could use a refactor

```julia
for indexset in (symbolnames, textnames, 1:5)
     for n in indexset
           @test all_the_things(n)
     end
end
```